### PR TITLE
feat: add module that verifies NCBI settings

### PR DESCRIPTION
**Pull request by Midnighter**
This is a small module that verifies an existing NCBI settings file or generates one suitable for prefetch and fasterq-dump. I aim to create a subworkflow as shown in https://github.com/Midnighter/sratools-settings-poc/blob/main/subworkflows/sra_dump.nf of which this is an important part.

I wasn't super sure about the naming since this module uses existing tooling but doesn't exactly implement a specific command. Maybe it belongs into `custom` instead?

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by Midnighter** [comment]
This module does not require any input and I think the other linting warnings/errors are again due to using a template.

**Review by drpatelh — APPROVED** [review]
Nice!

**Comment by drpatelh** [comment]
Tests are failing? 👀

**Comment by Midnighter** [comment]
Tried to fix conda, now others are failing 🙈

File: modules/custom/sratoolsncbisettings/main.nf [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+    output:
+    path('*.mkfg')     , emit: ncbi_settings
+    path 'versions.yml', emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    shell:
+    config = "/LIBS/GUID = \"${UUID.randomUUID().toString()}\"\\n/libs/cloud/report_instance_identity = \"true\"\\n"
```
Comment (drpatelh) [inline modules/custom/sratoolsncbisettings/main.nf:18]: Where is this `config` variable being used?

Comment (Midnighter) [reply modules/custom/sratoolsncbisettings/main.nf:18]: In the template.

Comment (drpatelh) [reply modules/custom/sratoolsncbisettings/main.nf:18]: Ah I see via `shell`.

**Comment by drpatelh** [comment]
Fine with me 🤷🏽

**Pull request merged**